docs/reference/builder.md

```diff
@@ -180,21 +180,28 @@ that set `abc` to `bye`.
 
 ### .dockerignore file
 
-If a file named `.dockerignore` exists in the root of `PATH`, then Docker
-interprets it as a newline-separated list of exclusion patterns. Docker excludes
-files or directories relative to `PATH` that match these exclusion patterns. If
-there are any `.dockerignore` files in `PATH` subdirectories, Docker treats
-them as normal files. 
-
-Filepaths in `.dockerignore` are absolute with the current directory as the
-root. Wildcards are allowed but the search is not recursive. Globbing (file name
-expansion) is done using Go's
-[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules.
-
-You can specify exceptions to exclusion rules. To do this, simply prefix a
-pattern with an `!` (exclamation mark) in the same way you would in a
-`.gitignore` file. Currently there is no support for regular expressions.
-Formats like `[^temp*]` are ignored. 
+If a file named `.dockerignore` exists at the root of the source
+repository, Docker interprets it as a newline-separated list of exclusion
```

Comment:
I see that *root of the source repository* isn't ideal.  But doesn't `.dockerignore` work even if you pass `URL` to `docker build` and not only if you pass `PATH`?  (I haven't actually tested that, but it would be logical.)